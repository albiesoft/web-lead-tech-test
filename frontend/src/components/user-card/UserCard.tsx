import type { FC } from "react";
import type { User } from "../../store/user.store.ts";

export interface UserCardProps {
  user: User;
}

const UserCard: FC<UserCardProps> = ({ user }) => {
  return (
    <div className="flex gap-2.5 items-center justify-center">
      <img
        className="w-10 h-auto rounded-full"
        src={user.profile}
        alt={user.name}
      />
      <div className="font-semibold">{user.name}</div>
    </div>
  );
};

export default UserCard;
