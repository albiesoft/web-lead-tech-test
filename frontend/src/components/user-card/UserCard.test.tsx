import { render } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import UserCard, { type UserCardProps } from "./UserCard";

describe("<UserCard>", () => {
  const defaultProps: UserCardProps = {
    user: {
      id: 1,
      name: "Alisha",
      profile: "https://randomuser.me/api/portraits/women/89.jpg",
    },
  };

  it("should match the snapshot", () => {
    const comp = render(<UserCard {...defaultProps} />);

    expect(comp).toMatchSnapshot();
  });
});
