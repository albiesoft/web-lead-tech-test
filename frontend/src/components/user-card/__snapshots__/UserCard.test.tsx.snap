// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<UserCard> > should match the snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex gap-2.5 items-center justify-center"
      >
        <img
          alt="Alisha"
          class="w-10 h-auto rounded-full"
          src="https://randomuser.me/api/portraits/women/89.jpg"
        />
        <div
          class="font-semibold"
        >
          Alisha
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex gap-2.5 items-center justify-center"
    >
      <img
        alt="Alisha"
        class="w-10 h-auto rounded-full"
        src="https://randomuser.me/api/portraits/women/89.jpg"
      />
      <div
        class="font-semibold"
      >
        Alisha
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
